server/api/utils/singletons/k8s.py

```diff
@@ -552,28 +552,22 @@ def ensure_configmap(
         labels: dict = None,
     ):
         namespace = self.resolve_namespace(namespace)
-
+        have_confmap = False
         label_name = "resource_name"
         full_name = f"{resource}-{name}"
-        name = (
-            full_name
-            if len(full_name) <= 63
-            else full_name[:59] + self._generate_rand_string(4)
-        )
 
-        have_confmap = False
-        configmaps_with_label = self.v1api.list_namespaced_config_map(
-            namespace=namespace, label_selector=f"{label_name}={full_name}"
+        configmap_with_label = self.get_configmap_for_function(
```

Comment:
this function is generic right now and not specifically bound to functions. This code changes those semantics

Reply:
@yanburman agreed! Changed naming